src/main/java/org/clx/library/dto/AuthorRequest.java

```diff
@@ -9,19 +13,29 @@
 @AllArgsConstructor
 @Builder
 public class AuthorRequest {
+    @NotEmpty(message = "name should not be empty")
+    @NotBlank
     private String name;
+    @NotEmpty(message = "name should not be empty")
+    @Column(unique = true)
+    @Email
     private String email;
-    private int age;
+    @NotBlank
+    @NotEmpty(message = "country shold not be empty")
     private String country;
+    private int age;
 
 
-
-    public Author authorRequestToAuthor(AuthorRequest authorRequest){
-        Author author= new Author();
+    public Author authorRequestToAuthor() {
+        Author author = new Author();
         author.setName(name);
         author.setEmail(email);
         author.setAge(age);
         author.setCountry(country);
         return author;
     }
+
+    public AuthorRequest authorToAuthorRequest(Author author) {
+        return AuthorRequest.builder().name(author.getName()).age(author.getAge()).country(author.getCountry()).email(author.getEmail()).build();
```

Comment:
why same class required to create to new object while it is under the same class object?

Reply:
so next time i am create another class for mapping.